test/topology_custom/test_raft_snapshot_truncation.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import pytest
+import time
+import logging
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for, wait_for_cql_and_get_hosts
+from test.topology.util import reconnect_driver, trigger_snapshot, get_topology_coordinator
+from test.pylib.rest_client import inject_error_one_shot
+from test.pylib.random_tables import RandomTables
+
+logger = logging.getLogger(__name__)
+
+
+async def get_raft_log_size(cql, host) -> int:
+    query = "select count(\"index\") from system.raft"
+    return (await cql.run_async(query, host=host))[0][0]
+
+
+async def get_raft_snap_id(cql, host) -> str:
+    query = "select snapshot_id from system.raft limit 1"
+    return (await cql.run_async(query, host=host))[0].snapshot_id
```

Comment:
These functions are already present in `test_raft_snapshot_request.py`. You can move them to `test/topology/util.py` if you want. I don't think it's necessary, though.